utility/block.go

```diff
@@ -110,13 +114,13 @@ func (u *UtilityContext) ApplyBlock() (string, error) {
 		//             Or wait until the entire lifecycle is over to evaluate an 'invalid' block
 
 		// Validate and apply the transaction to the Postgres database
-		txResult, err := u.ApplyTransaction(index, tx)
+		txResult, err := u.applyTransaction(index, tx)
 		if err != nil {
 			return "", err
 		}
 
-		if err := u.Context.IndexTransaction(txResult); err != nil {
-			u.logger.Fatal().Err(err).Msg("TODO(#327): We can apply the transaction but not index it. Crash the process for now")
+		if err := u.persistenceContext.IndexTransaction(txResult); err != nil {
```

Comment:
Here too `IndexTransaction` and not `IndexTx`? It makes me wonder if it's the same thing (If I use a bit of empathy, like if I am reading this for the first time)